OpenTelemetry (S3M) Collector Docs and Getting Started revisions (#1872)
SumoLogic/sumologic-documentation: docs/send-data/opentelemetry-collector/install-collector-on-macos.md

---
id: install-collector-on-macos

> title: Installing Collector on Mac OS
> sidebar_label: Install Collector on MacOS

Reviewer: ```suggestion
title: Installing Collector on macOS
sidebar_label: Install Collector on macOS
```

* [Manual step-by-step installation](#manual-step-by-step-installation)
* [UI Installation via App Catalog](#ui-installation-via-app-catalog)

### Install Script

#### Get the Installation token

Get your [installation token](/docs/manage/security/installation-tokens) if you don't have it already and assign it to an environment variable:

```bash
export SUMOLOGIC_INSTALLATION_TOKEN=<TOKEN>
```

#### Run the installation script

You can run the script in two ways:

* by piping `curl` straight into `bash`:

   ```bash
   sudo curl -Ls https://github.com/SumoLogic/sumologic-otel-collector/releases/latest/download/install.sh | sudo -E bash -s -- -d
   ```

* by first downloading the script, inspecting its contents for security, and then running it:

   ```bash
   curl -Lso install-otelcol-sumo.sh https://github.com/SumoLogic/sumologic-otel-collector/releases/latest/download/install.sh
   sudo -E bash ./install-otelcol-sumo.sh -d
   ```

The `-E` argument to `sudo` is needed to preserve the `SUMOLOGIC_INSTALLATION_TOKEN` environment variable in `sudo` session.

The `-d` argument to the script is needed on macOS. It will skip Systemd installation, which is not supported on macOS.

> It is going to perform the following operations:

Author: ```suggestion
It will perform the following operations:
```

---
id: install-collector-on-macos
title: Installing Collector on macOS
sidebar_label: Install Collector on macOS

> description: Learn how to install Collector

Reviewer: ```suggestion
description: Learn how to install the Sumo Logic OpenTelemetry Collector on macOS.
```

---

import useBaseUrl from '@docusaurus/useBaseUrl';

Follow the steps in this topic to install or uninstall an OpenTelemetry Collector on macOS. See [OpenTelemetry Collector](/docs/send-data/opentelemetry-collector) for information on other operating systems.

## System Requirements​

The Sumo Logic OpenTelemetry Collector is supported on both amd64 and arm64 architectures.

Minimal resource requirements are the following:

* 200 MB of disk space
* 64 MB of RAM

## Install

You can install our OpenTelemetry Collector using either of the following methods:

* [Install script](#install-script)
* [Manual step-by-step installation](#manual-step-by-step-installation)

### Install Script

#### Get the Installation token

> Get your [installation token](https://help.sumologic.com/docs/manage/security/installation-tokens) if you don't have it already and assign it to an environment variable:

Reviewer: ```suggestion
Get your [installation token](/docs/manage/security/installation-tokens) if you don't have it already and assign it to an environment variable:
```

   curl -Lso install-otelcol-sumo.sh https://github.com/SumoLogic/sumologic-otel-collector/releases/latest/download/install.sh
   sudo -E bash ./install-otelcol-sumo.sh -d
   ```

The `-E` argument to `sudo` is needed to preserve the `SUMOLOGIC_INSTALLATION_TOKEN` environment variable in `sudo` session.

The `-d` argument to the script is needed on macOS. It will skip Systemd installation, which is not supported on macOS.

It is going to perform the following operations:

* install or upgrade operation by placing the latest version as `/usr/local/bin/otelcol-sumo`,
* get [static configuration](https://github.com/SumoLogic/sumologic-otel-collector/blob/main/examples/sumologic.yaml) and place it as `/etc/otelcol-sumo/sumologic.yaml`
* create user configuration directory (`/etc/otelcol-sumo/conf.d`) with `common.yaml` file which will contain installation token

#### Script Options

The following arguments can be passed to the script:

| long name                   | short name | description                                                                                                                                                                  | takes value                |

> |-----------------------------|------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------|

Reviewer: ```suggestion
|:----------------------------|:------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:----------------------------|
```

#### Step 1. Download the binary

Examples for OpenTelemetry Collector version `0.73.0-sumo-0`.

##### amd64 (x86-64)

```bash
curl -sLo otelcol-sumo "https://github.com/SumoLogic/sumologic-otel-collector/releases/download/v0.73.0-sumo-0/otelcol-sumo-0.73.0-sumo-0-darwin_amd64"
```

##### arm64 (Apple Silicon)

```bash
curl -sLo otelcol-sumo "https://github.com/SumoLogic/sumologic-otel-collector/releases/download/v0.73.0-sumo-0/otelcol-sumo-0.73.0-sumo-0-darwin_arm64"
```

#### Step 2. Move the binary to your `PATH` environment

Move the downloaded binary into a directory from your `PATH` environment, so that it can be used by simply invoking `otelcol-sumo`.

```bash
chmod +x otelcol-sumo
sudo mv otelcol-sumo /usr/local/bin/otelcol-sumo
```

#### Step 3. Verify the Installation

To verify installation, run OpenTelemetry Collector.

```bash
otelcol-sumo --version
```

#### Step 4. Running Binary

As for now we do not support installation of OpenTelemetry Collector as service for macOS. It needs to be run manually.

```bash
sudo otelcol-sumo --config=/etc/otelcol-sumo/sumologic.yaml --config "glob:/etc/otelcol-sumo/conf.d/*.yaml"
```

### UI Installation via App Catalog

> 1. Go to app catalog and select `macOS - OpenTelemetry application`
> 
>    <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-mac-os.png')} alt="macOS in app catalog" />
> 
> 1. Select `Add New Collector` and click `Next`
> 
>    <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-macos-set-up-collector.png')} alt="set up collector" />
> 
> 1. Select installation token and customise your tags
> 
>    <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-macos-add-new-collector.png')} alt="add new collector" />
> 
> 1. Copy command and execute it in your system terminal
> 
>    <img src={useBaseUrl('img/send-data/opentelemetry-collector/macos-terminal.png')} alt="execute command in terminal" />
> 
> 1. Wait for installation to be completed
> 
>    <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-macos-success-registration.png')} alt="collector successfully registered" />
> 
> 1. Customize source configuration, download it, place it in `/etc/otelcol-sumo/conf.d` and then restart collector manually
> 
>    <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-macos-source-creation.png')} alt="source customisation" />
> 
> 1. Wait for finishing the installation
> 
>    <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-macos-success-installation.png')} alt="application installed successfully" />

Reviewer: Close gaps and use U.S. (not U.K.) spelling

```suggestion
1. Go to app catalog and select `macOS - OpenTelemetry application`.<br/> <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-mac-os.png')} alt="macOS in app catalog" />
1. Select `Add New Collector` and click `Next`.<br/> <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-macos-set-up-collector.png')} alt="set up collector" />
1. Select installation token and customize your tags.<br/> <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-macos-add-new-collector.png')} alt="add new collector" />
1. Copy command and execute it in your system terminal.<br/><img src={useBaseUrl('img/send-data/opentelemetry-collector/macos-terminal.png')} alt="execute command in terminal" />
1. Wait for installation to be completed.<br/> <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-macos-success-registration.png')} alt="collector successfully registered" />
1. Customize source configuration, download it, place it in `/etc/otelcol-sumo/conf.d` and then restart collector manually.<br/> <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-macos-source-creation.png')} alt="source customization" />
1. Wait for finishing the installation.<br/> <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-macos-success-installation.png')} alt="application installed successfully" />
```

### Additional settings

> This section describes common OpenTelemetry customisations:

Reviewer: ```suggestion
This section describes common OpenTelemetry customizations:
```
